File: src/main/java/com/nao4j/subtrol/controller/CostController.java
package com.nao4j.subtrol.controller;

import com.nao4j.subtrol.document.internal.Costs;
import com.nao4j.subtrol.document.internal.ExactPeriod;
import com.nao4j.subtrol.dto.UserCredentials;
import com.nao4j.subtrol.service.CostService;
import lombok.RequiredArgsConstructor;
import org.springframework.format.annotation.DateTimeFormat;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.time.LocalDateTime;

import static org.springframework.format.annotation.DateTimeFormat.ISO.DATE_TIME;

@RestController
@RequiredArgsConstructor
@RequestMapping("/api/costs")
public class CostController {

    private final CostService costService;

    @GetMapping
    public Costs getAllInPeriod(
            @RequestParam @DateTimeFormat(iso = DATE_TIME) final LocalDateTime start,
            @RequestParam @DateTimeFormat(iso = DATE_TIME) final LocalDateTime end
    ) {
        final var user = (UserCredentials) SecurityContextHolder.getContext().getAuthentication().getDetails();
        return costService.calculateForPeriod(user.getId(), new ExactPeriod(start, end));
    }

}

File: src/main/kotlin/com/nao4j/subtrol/controller/CostController.kt
package com.nao4j.subtrol.controller

import com.nao4j.subtrol.dto.UserCredentials
import com.nao4j.subtrol.model.Costs
import com.nao4j.subtrol.model.ExactPeriod
import com.nao4j.subtrol.service.CostService
import org.springframework.format.annotation.DateTimeFormat
import org.springframework.format.annotation.DateTimeFormat.ISO.DATE_TIME
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController
import java.time.LocalDateTime

@RestController
@RequestMapping("/api/costs")
class CostController(private val costService: CostService) {

    @GetMapping
    fun getAllInPeriod(
            @RequestParam @DateTimeFormat(iso = DATE_TIME) start: LocalDateTime,
            @RequestParam @DateTimeFormat(iso = DATE_TIME) end: LocalDateTime
    ): Costs {
        val user = SecurityContextHolder.getContext().authentication.details as UserCredentials
        return costService.calculateForPeriod(user.id, ExactPeriod(start, end))
    }

}
